docs/shared/recipes/vite/vite-inferred-targets.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Vite Inferred Targets
+description: This document explains how to enable inferred targets for Vite projects.
+---
+
+# Vite Inferred Targets
+
+In Nx version 17.3, the `@nx/vite` plugin can create [inferred targets](/concepts/inferred-targets) for projects that have a Vite configuration file present. This means you can run `nx build my-project`, `nx serve my-project`, `nx preview my-project`, `nx serve-static my-project` and `nx test my-project` for that project, even if there is no `build`, `serve`, `preview`, `serve-static` or `test` targets defined in `package.json` or `project.json`.
+
+## Setup
+
+To enable inferred targets, add `@nx/vite/plugin` to the `plugins` array in `nx.json`.
+
+```json {% fileName="nx.json" %}
+{
+  "plugins": [
+    {
+      "plugin": "@nx/vite/plugin",
+      "options": {
+        "buildTargetName": "build",
+        "previewTargetName": "preview",
+        "testTargetName": "test",
+        "serveTargetName": "serve",
+        "serveStaticTargetName": "serve-static"
+      }
+    }
+  ]
+}
+```
+
+## Target Inference Process
+
+### Identify Valid Projects
+
+The `@nx/vite/plugin` plugin will create a target for any project that has a Vite configuration file present. Any of the following files will be recognized as a Vite configuration file:
+
+- `vite.config.js`
+- `vite.config.ts`
+- `vite.config.mjs`
+- `vite.config.mts`
+- `vite.config.cjs`
+- `vite.config.cts`
+- `vitest.config.js`
+- `vitest.config.ts`
+- `vitest.config.mjs`
+- `vitest.config.mts`
+- `vitest.config.cjs`
+- `vitest.config.cts`
+
+### Name the Inferred Target
+
+Once a Vite configuration file has been identified, the targets are created with the name you specify under `buildTargetName`, `serveTargetName`, `previewTargetName`, `serveStaticTargetName` or `testTargetName` in the `nx.json` `plugins` array. The default names for the inferred targets are `build`, `serve`, `preview`, `serve-static` and `test`.
+
+### Set Inferred Options
+
+The `@nx/vite/plugin` plugin will start with any `targetDefaults` you have set in `nx.json`, and then apply the following properties:
+
+#### For `build`
+
+| Property  | Sample Value                           | Description                                                               |
+| --------- | -------------------------------------- | ------------------------------------------------------------------------- |
+| `command` | `vite build`                           | Build the project                                                         |
+| `cache`   | `true`                                 | Automatically cache the task                                              |
+| `inputs`  | `production`                           | Break the cache if any file in the project changes                        |
```

Comment:
`production` breaks the cache if non-test files change.  `default` would break the cache if any file in the project changes.